test/functional/plugin/lsp_spec.lua

```diff
@@ -926,6 +926,49 @@ describe('LSP', function()
         end;
       }
     end)
+
+    it('should handle return nil when decoding a list', function()
```

Comment:
@mjlbach sorry to be a bother but in the interest of time/trying to get this in for 0.5 I wonder if you could help. The testing helpers for the lsp presume a much better understanding of neovim's client than I have and I'd like to avoid losing too much time trying to wrangling with it.

I'm essentially just trying to send a request back and forth from the fake server and assert that `vim.NIL` is kept if the params contains a list but my test keeps failing because the expected callbacks are in the wrong order 🤔 . There's kind of a lot going on across the various test helpers

Is this the kind of test you had in mind, should it be something simpler that just checks the behaviour of `handle_body`?

Reply:
I can help debug any tests this weekend. I would try to match the callbacks expected by other tests, there shouldn't be anything unique about this other than the fact that you're using code-actions which has 2 round trips

Reply:
@mjlbach I've spent more time plugging away at this now and I'm not really sure how specifically to test this based on the examples in `lsp_spec.lua`. It doesn't have any docs so it's mostly been trial and error/reading through some of the helper implementations.

What should the flow be within my test case be? I tried basing it on the issue I saw i.e. I've tried triggering and responding to a code action which should trigger an `executeCommand` if this command contained a list with a `nil` value then if this change was working we'd expect it to get through to the test in `fake_lsp_server.lua` with the `vim.NIL` intact?

Trying to trigger the `vim.lsp.buf.code_action()` doesn't seem to do anything i.e. the `textDocument/codeAction` isn't sent to the fake lsp server and none of the subsequent commands are executed, I also can't tell whether it getting through would correctly validate that `NIL` was no longer being filtered out. Just passing notifications through using `client.notify` doesn't seems to make a difference whether I comment out my changes or not.